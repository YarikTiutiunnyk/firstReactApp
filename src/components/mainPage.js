import React from 'react';

import {connect} from 'react-redux'
import { Container, Image, Button, Segment} from 'semantic-ui-react'
import { Switch, Route, useRouteMatch} from "react-router-dom";
import logoSite from '../img/logoImg.png'

import { setCollections, setFilms, addFilms, setFilmsError, setSeries, addSeries, setSeriesError } from '../actions/film';
import { setSerchFilmsString, setSerchSeriesString} from '../actions/serch';
import { serchOMDBFilms } from './serch'

import CardList from './cardList';
import Menu from './menu';
import SerchBlock from './serch';


function mainPage (props) {
  //let match = useRouteMatch();

  return (
    <Container>
        <Image centered src= {logoSite}   />
        <Menu />     
        <Switch>  
          <Route exact path="/films">
            <SerchBlock itemFound= {props.filmObj.found} textError= {props.filmObj.textError} setItems={props.setFilms} setItemsError={props.setFilmsError} setSerchString={props.setSerchFilmsString} type='movie'/>
            <SerchFilmPage  filmObj={props.filmObj} setFilms={props.addFilms} setFilmsError={props.setFilmsError} serchStringFilms={props.serchStringFilms}/>
          </Route>
          <Route exact path="/series">
            <SerchBlock itemFound= {props.seriesObj.found} textError= {props.seriesObj.textError} setItems={props.setSeries} setItemsError={props.setSeriesError} setSerchString={props.setSerchSeriesString} type='series'/>
            <SerchSeriesPage filmObj={props.seriesObj} setFilms={props.addSeries} setFilmsError={props.setSeriesError} serchStringFilms={props.serchStringSeries}/>
          </Route>
          <Route exact path="/">
            <CardList films={props.masCollection} currentURL='/collection'/>
          </Route>

        </Switch> 
           
      </Container>
  );
}

//Функція зв'язує redux state з props 
const mapStateToProps = state => {
  //Фільми з колекції які завантажуються статично
  const masCollection = state.filmColection.collection.mas;
  //Фільми які завантажуються динамічно
  const filmObj = state.filmColection.films;
  const serchStringFilms = state.serch.serchStringFilms;
  //Серіали які завантажуються динамічно
  const seriesObj = state.filmColection.series;
  const serchStringSeries = state.serch.serchStringSeries;

  return { masCollection, filmObj, serchStringFilms, seriesObj, serchStringSeries};
};
  
  
//Функція зв'язує actions з props 
const mapDispatchToProps = dispatch=>({
  setCollections : films => dispatch(setCollections(films)),

  setFilms : objectWithFilms => dispatch(setFilms(objectWithFilms)),
  addFilms: objectWithFilms => dispatch(addFilms(objectWithFilms)),
  setFilmsError : strError => dispatch(setFilmsError(strError)),
  
  setSeries : objectWithSeries => dispatch(setSeries(objectWithSeries)), 
  addSeries : objectWithSeries => dispatch(addSeries(objectWithSeries)), 
  setSeriesError : strError => dispatch(setSeriesError(strError)),
  
  setSerchFilmsString : str => dispatch(setSerchFilmsString(str)),
  setSerchSeriesString : str => dispatch(setSerchSeriesString(str)),

})

export default connect(mapStateToProps, mapDispatchToProps)(mainPage);





function SerchFilmPage (props) {
let match = useRouteMatch();

function handleClick(e) {
  e.preventDefault();
  console.log('По ссылке кликнули.');
  serchOMDBFilms(props.serchStringFilms, props.setFilms, props.setFilmsError, Math.ceil(props.filmObj.count/10)+1 )
}

return (

    <div>
        <CardList films={props.filmObj.mas} currentURL={`${match.url}`}/>
        <Segment basic textAlign='center'>
          <Button onClick={handleClick} disabled={props.filmObj.count >= props.filmObj.found }  primary size='huge' style={{ marginTop: '1.5em', minWidth: '10em', marginBottom: '50px'}}> More  </Button>
        </Segment>
    </div>
);
}


function SerchSeriesPage (props) {
  let match = useRouteMatch();
  
  function handleClick(e) {
    e.preventDefault();
    console.log('По ссылке кликнули.');
    console.log(props);

    serchOMDBFilms(props.serchStringFilms, props.setFilms, props.setFilmsError, Math.ceil(props.filmObj.count/10)+1, 'series' )
  }
  
  return (
  
      <div>
          <CardList films={props.filmObj.mas} currentURL={`${match.url}`}/>
          <Segment basic textAlign='center'>
            <Button onClick={handleClick} disabled={props.filmObj.count >= props.filmObj.found }  primary size='huge' style={{ marginTop: '1.5em', minWidth: '10em', marginBottom: '50px'}}> More  </Button>
          </Segment>
      </div>
  );
  }



/*class SerchSeriesPage extends React.Component {
    
    render(){
        return (
          <div>
            <h2>SerchSeries</h2>   
          </div>
        );
    }
}*/

