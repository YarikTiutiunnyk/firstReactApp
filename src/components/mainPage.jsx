import React from 'react';

import { connect } from 'react-redux';
import {
  Container,
  Image,
  Button,
  Segment,
  Dimmer,
  Loader
} from 'semantic-ui-react';
import { Switch, Route, useRouteMatch } from 'react-router-dom';
import PropTypes from 'prop-types';
import logoSite from '../img/logoImg.png';

import {
  setCollections,
  setFilms,
  addFilms,
  setFilmsError,
  setSeries,
  addSeries,
  setSeriesError,
  itemsFetchData
} from '../actions/film';
import { setSerchFilmsString, setSerchSeriesString } from '../actions/serch';

import CardList from './cardList';
import Menu from './menu';
import SerchBlock from './serch';

export function mainPage(props) {
  //let match = useRouteMatch();

  return (
    <Container>
      <Image centered src={logoSite} />
      <Menu />
      <Switch>
        <Route exact path="/firstReactApp/films">
          <SerchBlock
            itemFound={props.filmObj.found}
            textError={props.filmObj.textError}
            setItems={props.setFilms}
            setItemsError={props.setFilmsError}
            type="movie"
          />
          <SerchFilmPage
            filmObj={props.filmObj}
            setFilms={props.addFilms}
            setFilmsError={props.setFilmsError}
            serchStringFilms={props.serchStringFilms}
            fetchData={props.fetchData}
          />
        </Route>
        <Route exact path="/firstReactApp/series">
          <SerchBlock
            itemFound={props.seriesObj.found}
            textError={props.seriesObj.textError}
            setItems={props.setSeries}
            setItemsError={props.setSeriesError}
            type="series"
          />
          <SerchSeriesPage
            filmObj={props.seriesObj}
            setFilms={props.addSeries}
            setFilmsError={props.setSeriesError}
            serchStringFilms={props.serchStringSeries}
            fetchData={props.fetchData}
          />
        </Route>
        <Route exact path="/firstReactApp/">
          <CardList
            films={props.masCollection}
            currentURL="/firstReactApp/collection"
          />
        </Route>
      </Switch>
    </Container>
  );
}
mainPage.propTypes = {
  filmObj: PropTypes.object.isRequired,
  setFilms: PropTypes.func.isRequired,
  setFilmsError: PropTypes.func.isRequired,
  setSerchFilmsString: PropTypes.func.isRequired,
  addFilms: PropTypes.func.isRequired,
  serchStringFilms: PropTypes.string.isRequired,

  seriesObj: PropTypes.object.isRequired,
  setSeries: PropTypes.func.isRequired,
  setSeriesError: PropTypes.func.isRequired,
  setSerchSeriesString: PropTypes.func.isRequired,
  addSeries: PropTypes.func.isRequired,
  serchStringSeries: PropTypes.string.isRequired,

  masCollection: PropTypes.array.isRequired
};

//Функція зв'язує redux state з props
const mapStateToProps = (state) => {
  //Фільми з колекції які завантажуються статично
  const masCollection = state.filmColection.collection.mas;
  //Фільми які завантажуються динамічно
  const filmObj = state.filmColection.films;
  const serchStringFilms = state.serch.serchStringFilms;
  //Серіали які завантажуються динамічно
  const seriesObj = state.filmColection.series;
  const serchStringSeries = state.serch.serchStringSeries;

  return {
    masCollection,
    filmObj,
    serchStringFilms,
    seriesObj,
    serchStringSeries
  };
};

//Функція зв'язує actions з props
const mapDispatchToProps = (dispatch) => ({
  setCollections: (films) => dispatch(setCollections(films)),

  setFilms: (objectWithFilms) => dispatch(setFilms(objectWithFilms)),
  addFilms: (objectWithFilms) => dispatch(addFilms(objectWithFilms)),
  setFilmsError: (strError) => dispatch(setFilmsError(strError)),

  setSeries: (objectWithSeries) => dispatch(setSeries(objectWithSeries)),
  addSeries: (objectWithSeries) => dispatch(addSeries(objectWithSeries)),
  setSeriesError: (strError) => dispatch(setSeriesError(strError)),

  setSerchFilmsString: (str) => dispatch(setSerchFilmsString(str)),
  setSerchSeriesString: (str) => dispatch(setSerchSeriesString(str)),
  fetchData: (name, setItems, setItemsError, page, type) =>
    dispatch(itemsFetchData(name, setItems, setItemsError, page, type))
});

export default connect(mapStateToProps, mapDispatchToProps)(mainPage);

export function SerchFilmPage(props) {
  let match = useRouteMatch();

  function handleClick(e) {
    e.preventDefault();
    //console.log('По ссылке кликнули.');
    props.fetchData(
      props.serchStringFilms,
      props.setFilms,
      props.setFilmsError,
      Math.ceil(props.filmObj.count / 10) + 1
    );
  }

  return (
    <div>
      <Segment basic>
        <Dimmer active={props.filmObj.isLoading} inverted>
          <Loader inverted>Loading</Loader>
        </Dimmer>
        <CardList films={props.filmObj.mas} currentURL={`${match.url}`} />
      </Segment>

      <Segment basic textAlign="center">
        <Button
          onClick={handleClick}
          disabled={props.filmObj.count >= props.filmObj.found}
          primary
          size="huge"
          style={{ marginTop: '0.5em', minWidth: '10em', marginBottom: '50px' }}
        >
          {' '}
          More{' '}
        </Button>
      </Segment>
    </div>
  );
}
SerchFilmPage.propTypes = {
  filmObj: PropTypes.shape({
    mas: PropTypes.array,
    count: PropTypes.number,
    found: PropTypes.string,
    isLoading: PropTypes.bool
  }),
  serchStringFilms: PropTypes.string.isRequired,
  setFilms: PropTypes.func.isRequired,
  setFilmsError: PropTypes.func.isRequired
};

export function SerchSeriesPage(props) {
  let match = useRouteMatch();

  function handleClick(e) {
    e.preventDefault();
    //console.log('По ссылке кликнули.');
    props.fetchData(
      props.serchStringFilms,
      props.setFilms,
      props.setFilmsError,
      Math.ceil(props.filmObj.count / 10) + 1,
      'series'
    );
  }

  return (
    <div>
      <Segment basic>
        <Dimmer active={props.filmObj.isLoading} inverted>
          <Loader inverted>Loading</Loader>
        </Dimmer>
        <CardList films={props.filmObj.mas} currentURL={`${match.url}`} />
      </Segment>

      <Segment basic textAlign="center">
        <Button
          onClick={handleClick}
          disabled={props.filmObj.count >= props.filmObj.found}
          primary
          size="huge"
          style={{ marginTop: '0.5em', minWidth: '10em', marginBottom: '50px' }}
        >
          {' '}
          More{' '}
        </Button>
      </Segment>
    </div>
  );
}
SerchSeriesPage.propTypes = {
  filmObj: PropTypes.shape({
    mas: PropTypes.array,
    count: PropTypes.number,
    found: PropTypes.string,
    isLoading: PropTypes.bool
  }),
  serchStringFilms: PropTypes.string.isRequired,
  setFilms: PropTypes.func.isRequired,
  setFilmsError: PropTypes.func.isRequired
};
