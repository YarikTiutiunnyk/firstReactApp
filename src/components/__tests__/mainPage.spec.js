import React from 'react';
import { shallow } from 'enzyme';
import { mainPage as MainPage } from '../mainPage';

describe('MainPage test', () => {
  const mocksetFilms = jest.fn();
  const mocksetFilmsError = jest.fn();
  const mockSerchFilmsString = jest.fn();
  const mockaddFilms = jest.fn();

  const mocksetSeries = jest.fn();
  const mocksetSeriesError = jest.fn();
  const mockSerchSeriesString = jest.fn();
  const mockaddSeries = jest.fn();

  const props = {
    filmObj: {
      found: '1',
      count: 1,
      mas: [
        {
          Title: 'Name 1',
          Year: '2001',
          imdbID: 'tt0259711',
          Type: 'movie',
          Poster: 'URL'
        }
      ],
      textError: 'Error film'
    },
    setFilms: mocksetFilms,
    setFilmsError: mocksetFilmsError,
    setSerchFilmsString: mockSerchFilmsString,
    addFilms: mockaddFilms,
    serchStringFilms: 'filmName',

    seriesObj: {
      found: '1',
      count: 1,
      mas: [
        {
          Title: 'Name 2',
          Year: '2001',
          imdbID: 'tt0259711',
          Type: 'series',
          Poster: 'URL'
        }
      ],
      textError: 'Error series'
    },
    setSeries: mocksetSeries,
    setSeriesError: mocksetSeriesError,
    setSerchSeriesString: mockSerchSeriesString,
    addSeries: mockaddSeries,
    serchStringSeries: 'seriesName',

    masCollection: [
      {
        Title: 'Name 3',
        Year: '2001',
        imdbID: 'tt0259711',
        Type: 'movie',
        Poster: 'URL'
      }
    ]
  };

  describe('MainPage initial', () => {
    const wrapper = shallow(<MainPage {...props} />);
    //console.log(wrapper.debug());

    it('Collections actibve renders properly', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });

  /*describe('SerchFilmPage tests', () => {
    const propsForSerchFilmPage = {
      filmObj: props.filmObj,
      serchStringFilms: props.serchStringFilms,
      setFilms: props.setFilms,
      setFilmsError: props.setFilmsError
    };
    const wrapper = mount(<SerchFilmPage {...propsForSerchFilmPage} />);
    console.log(wrapper.debug());

    it('Collections actibve renders properly', () => {
      //expect(wrapper).toMatchSnapshot();
    });
  });*/
});
