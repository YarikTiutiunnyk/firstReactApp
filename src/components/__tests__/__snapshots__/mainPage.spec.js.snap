// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainPage test MainPage initial Collections actibve renders properly 1`] = `
<Container>
  <Image
    as="img"
    centered={true}
    src="logoImg.png"
    ui={true}
  />
  <Connect(MenuExampleBasic) />
  <Switch>
    <Route
      exact={true}
      path="/firstReactApp/films"
    >
      <InputExampleActionIconButton
        itemFound="1"
        setItems={[MockFunction]}
        setItemsError={[MockFunction]}
        setSerchString={[MockFunction]}
        textError="Error film"
        type="movie"
      />
      <SerchFilmPage
        filmObj={
          Object {
            "count": 1,
            "found": "1",
            "mas": Array [
              Object {
                "Poster": "URL",
                "Title": "Name 1",
                "Type": "movie",
                "Year": "2001",
                "imdbID": "tt0259711",
              },
            ],
            "textError": "Error film",
          }
        }
        serchStringFilms="filmName"
        setFilms={[MockFunction]}
        setFilmsError={[MockFunction]}
      />
    </Route>
    <Route
      exact={true}
      path="/firstReactApp/series"
    >
      <InputExampleActionIconButton
        itemFound="1"
        setItems={[MockFunction]}
        setItemsError={[MockFunction]}
        setSerchString={[MockFunction]}
        textError="Error series"
        type="series"
      />
      <SerchSeriesPage
        filmObj={
          Object {
            "count": 1,
            "found": "1",
            "mas": Array [
              Object {
                "Poster": "URL",
                "Title": "Name 2",
                "Type": "series",
                "Year": "2001",
                "imdbID": "tt0259711",
              },
            ],
            "textError": "Error series",
          }
        }
        serchStringFilms="seriesName"
        setFilms={[MockFunction]}
        setFilmsError={[MockFunction]}
      />
    </Route>
    <Route
      exact={true}
      path="/firstReactApp/"
    >
      <CardList
        currentURL="/firstReactApp/collection"
        films={
          Array [
            Object {
              "Poster": "URL",
              "Title": "Name 3",
              "Type": "movie",
              "Year": "2001",
              "imdbID": "tt0259711",
            },
          ]
        }
      />
    </Route>
  </Switch>
</Container>
`;
